packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -461,11 +464,41 @@ export default function useAutocomplete(props) {
     },
   );
 
+  const checkHighlightedOptionExists = () => {
+    if (
+      highlightedIndexRef.current !== -1 &&
+      previousProps.filteredOptions &&
+      previousProps.filteredOptions.length !== filteredOptions.length &&
+      (multiple
+        ? previousProps.value.every((val, i) => getOptionLabel(value[i]) === getOptionLabel(val))
```

Comment:
This PR introduces a runtime error `TypeError: Cannot read properties of undefined (reading 'label')` at `getOptionLabel(value[i])` when under these conditions:
1) AutoComplete is using the `multiple` flag
2) There is currently one selected option
3) You delete the existing option while the dropdown is open

When you delete the option, `value` is now `[]` and so `value[i]` is undefined. When fed to `getOptionLabel`, it crashes because it expects option to not be undefined. (`getOptionLabel: getOptionLabelProp = (option) => option.label ?? option`)